pkg/client/cache/reflector.go

```diff
@@ -160,16 +164,50 @@ var (
 // required, and a cleanup function.
 func (r *Reflector) resyncChan() (<-chan time.Time, func() bool) {
 	if r.resyncPeriod == 0 {
+		r.nextResync = time.Time{}
 		return neverExitWatch, func() bool { return false }
 	}
 	// The cleanup function is required: imagine the scenario where watches
 	// always fail so we end up listing frequently. Then, if we don't
 	// manually stop the timer, we could end up with many timers active
 	// concurrently.
+	r.nextResync = time.Now().Add(r.resyncPeriod)
```

Comment:
We would generally turn time.Now() into a static package private function that we can override in unit tests to validate the behavior we expect w.r.t. time changes.